xapian-bindings/java/java.i

```diff
@@ -174,21 +174,63 @@ namespace Xapian {
     bool hasNext() const { return Xapian::iterator_valid(*self); }
 }
 
+%ignore MSetIterator::next();
+%javamethodmodifiers MSetIterator::next_helper() "private"
+
 %extend MSetIterator {
-    Xapian::docid next () {
-	Xapian::docid tmp;
+    void next_helper () {
 	if (Xapian::iterator_valid(*self)) {
```

Comment:
From my understanding we don't have any Java way of proceeding to the next element of the iterator, a helper function which makes a JNI call (which will execute  `++(*self)`) is needed to move to the next item.